src/plot_api/plot_api.js

```diff
@@ -2093,7 +2093,7 @@ Plotly.update = function update(gd, traceUpdate, layoutUpdate, traces) {
     // fill in redraw sequence
     var seq = [];
 
-    if(restyleFlags.fullReplot && relayoutFlags.layoutReplot) {
+    if(restyleFlags.fullReplot || relayoutFlags.layoutReplot) {
```

Comment:
No. `&&` looks correct here.

That if-else-block looks like:

```js
if(restyleFlags.fullReplot && relayoutFlags.layoutReplot) {
  // full restyle and relayout update
}
else if(restyleFlags.fullReplot) {
  // full restyle update (but not relayout update)
}
else if(relayoutFlags.layoutReplot) {
    // full relayout update (but not restyle update)
}
else {
  // other more specific updates.
}
```

Maybe the problem is in [`layoutReplot`](https://github.com/plotly/plotly.js/blob/3b85f7ff966293d0c4a2b56c6f8e8c744ee96e24/src/plot_api/subroutines.js#L275-L280)?